libc/test/src/stdio/CMakeLists.txt

```diff
@@ -220,6 +220,20 @@ add_libc_test(
     libc.src.stdio.vprintf
 )
 
+
+if(LLVM_LIBC_FULL_BUILD)
+  # In fullbuild mode, fscanf's tests use the internal FILE for other functions.
+  list(APPEND fscanf_test_deps
+       libc.src.stdio.fclose
+       libc.src.stdio.ferror
+       libc.src.stdio.fopen
+       libc.src.stdio.fwrite
+  )
+else()
+# Else in overlay mode they use the system's FILE.
+ set(fscanf_test_copts "-DLIBC_COPT_SCANF_USE_SYSTEM_FILE")
```

Comment:
This opt should likely be named as `LIBC_COPT_STDIO_USE_SYSTEM_FILE` and shared between `printf`, `scanf` and all other stdio pieces which can work with both system file or libc's own file.

Reply:
done. I renamed the flag in cmake to `use_system_file`, though you can change that in your printf config patch.